src/api/c/shift.cpp

```diff
@@ -18,12 +18,12 @@ using af::dim4;
 using namespace detail;
 
 template<typename T>
-static inline af_array shift(const af_array in, const af::dim4 &sdims)
+static inline af_array shift(const af_array in, const int sdims[4])
 {
     return getHandle(shift<T>(getArray<T>(in), sdims));
 }
 
-af_err af_shift(af_array *out, const af_array in, const af::dim4 &sdims)
+af_err af_shift(af_array *out, const af_array in, const int sdims[4])
```

Comment:
No its still a pointer. It just looks like it is passing by value. The [4] is irrelevant.